### PR TITLE
Remove requirements_test.txt

### PR DESCRIPTION
**Pull request by schmmd**
Partially addresses https://github.com/allenai/allennlp/issues/1378

### CODE REVIEW

**Comment by schmmd** [comment]
This also remove `scripts/install_requirements.sh`.  We no longer need to manually install spacy due to https://github.com/allenai/allennlp/pull/933 and I couldn't see where we're using `punkt`.  I suspect, however, that we still need `punkt` based on https://github.com/allenai/allennlp/pull/44--so the build will probably fail and I'll revert the final commit.

File: scripts/install_requirements.sh [hunk 1]
```diff
@@ -1,10 +0,0 @@
-#!/bin/bash
-
-pip install -r requirements.txt
```
Comment (DeNeutoy) [inline scripts/install_requirements.sh]: Don't remove this script please - it's useful when installing into a virtual environment (e.g not docker). The `spacy download en_core_web_sm` is still required and I think the `punkt` data package from NLTK is used in the Stemmer still too.

Comment (DeNeutoy) [reply scripts/install_requirements.sh]: Oh right, you don't want to have to download the spacy model. Personally I don't like this approach (e.g it makes what packages are actually required kind of opaque if we download them at runtime) but if others think it is fine then sure.

Comment (schmmd) [reply scripts/install_requirements.sh]: I'm happy to leave the script and add `spacy download en_core_web_sm`--but I don't want to add `punkt` back unless I can identify where it's being used.  It's not being used in any of our test cases.  I wanted to remove both of those in this PR so I could run the tests in a clean environment.

Comment (schmmd) [reply scripts/install_requirements.sh]: It looks like we "use it" in the unused and untested:

```
$ git grep NltkWordSplitter
allennlp/data/tokenizers/word_splitter.py:class NltkWordSplitter(WordSplitter):
```

Comment (DeNeutoy) [reply scripts/install_requirements.sh]: It's used here: https://github.com/allenai/allennlp/blob/master/allennlp/data/tokenizers/word_splitter.py#L141

If you want to remove this tokeniser and remove this data download that's fine with me.

File: Dockerfile [hunk 2]
```diff
@@ -41,7 +41,6 @@ RUN apt-get install -y -t jessie-backports openjdk-8-jdk
 # Copy select files needed for installing requirements.
 # We only copy what we need here so small changes to the repository does not trigger re-installation of the requirements.
 COPY requirements.txt .
-COPY requirements_test.txt .
 COPY scripts/install_requirements.sh scripts/install_requirements.sh
 RUN INSTALL_TEST_REQUIREMENTS="true" ./scripts/install_requirements.sh
```
Comment (DeNeutoy) [inline Dockerfile]: This environment variable can be removed now.

File: scripts/install_requirements.sh (hunk 1 above)
Comment (schmmd) [reply scripts/install_requirements.sh]: I will in a separate PR.

**Pull request merged**